raven/src/main/java/org/opencadc/raven/RavenInitAction.java

```diff
@@ -235,4 +282,91 @@ static MultiValuedProperties getConfig() {
             throw new IllegalStateException("invalid config: failed to load SQLGenerator: " + cname);
         }
     }
+
+    private class AvailabilityCheck implements Runnable {
+
+        private Set<StorageSite> sites;
+        private Map<URI, SiteState> siteStaties;
+        private Map<URI, Availability> siteAvailabilities;
+        
+        public AvailabilityCheck(Set<StorageSite> sites) {
+            this.sites = sites;
+            this.siteStaties = new HashMap<>(sites.size());
+            this.siteAvailabilities = new HashMap<>(sites.size());
+            try {
+                Context initialContext = new InitialContext();
```

Comment:
Here you need to be careful to check and unbind a previously bound object. yeah, it can't actually happen in the normal way of running raven in a container and always killing that and starting as new one, but the java code should be usable in a normal tomcat container, which means reloading the app without a tomcat restart.

In would also be good to be able to unbind when then thread gets terminated to not leave junk around.

Reply:
Updated to check for and remove an existing bind before binding the current object, and unbind on terminate.

Reply:
Why this kind of check?  Why not just lookup and unbind? Why not just blindly unbind like up in the terminate method?